### DISCUSSION TITLE
Enhance embed with GDPR-compliant user consent mechanism

### DISCUSSION THREAD

**Discussion by deployn** [delucis/astro-embed#275] [Feature Requests]
I would love to see an improvement for e.g. the YouTube component to be GDPR compliant by implementing a user consent mechanism before loading any YouTube content or cookies.

## Current Behavior
- An image thumbnail of the YouTube video is displayed initially.
- Clicking the image loads the full YouTube embed.

## Proposed Changes
1. Keep the initial image thumbnail display.
2. Implement a consent modal that appears when a user first attempts to play a video.
3. Store user consent preference using localStorage.
4. Load YouTube content only after receiving user consent.

**Comment by anotheri** [comment]
This is one of the solutions for it: https://cookieconsent.orestbida.com/
